File: back/app/models/telescopestatus.py
from pydantic import BaseModel

class TelescopeInfo(BaseModel):
    current_task : str = 'IDLE'
    ra : float = 0.0
    dec : float = 0.0
    object: str = ''
    processing: bool = False
    stacking: bool = False
    exposition: float = 2.0
    last_error: int = 0
    ccd_orientation : float = 0.0
    stacked: int = 0
    discarded : int = 0

File: front/easyastro/lib/models/telescopestatus.dart
class TelescopeStatus {
  String currentTask;
  double ra;
  double dec;
  String object;
  bool stacking;
  double exposition;
  int lastError;
  double ccdOrientation;
  int stacked;
  int discarded;

  TelescopeStatus(
      this.currentTask,
      this.ra,
      this.dec,
      this.object,
      this.stacking,
      this.exposition,
      this.lastError,
      this.ccdOrientation,
      this.stacked,
      this.discarded);

  TelescopeStatus.fromJson(Map<String, dynamic> json)
      : currentTask = json['current_task'],
        ra = json['ra'],
        dec = json['dec'],
        object = json['object'],
        stacking = json['stacking'],
        exposition = json['exposition'],
        lastError = json['last_error'],
        ccdOrientation = json['ccd_orientation'],
        stacked = json['stacked'],
        discarded = json['discarded'];

  Map<String, dynamic> toJson() => {
        'current_task': currentTask,
        'ra': ra,
        'dec': dec,
        'object': object,
        'stacking': stacking,
        'exposition': exposition,
        'last_error': lastError,
        'ccd_orientation': ccdOrientation,
        'stacked': stacked,
        'discarded': discarded
      };
}
